src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -12341,8 +12354,11 @@ private bool ScanCast(bool forPattern = false)
                         SyntaxKind.PlusToken or
                         SyntaxKind.MinusToken or
                         SyntaxKind.AmpersandToken or
-                        SyntaxKind.AsteriskToken or
-                        SyntaxKind.DotDotToken
+                        SyntaxKind.AsteriskToken
+                            => true,
+
+                        // `(X)..` must be a cast of a range expression, not a member access of some arbitrary expression.
+                        SyntaxKind.DotToken when IsAtDotDotToken()
```

Comment:
Do we have a test affected by this logic? No test in Syntax.Unittests broke when I removed this line

Reply:
Positive tests added in RangeExpressionParsingTests.

Reply:
I both removed the `when IsAtDotDotToken()`, negated it, and removed the case entirely, but still couldn't find any break in `Syntax.UnitTests`.

Reply:
tests added.